mathesar/utils/datafiles.py

```diff
@@ -1,41 +1,44 @@
+from time import time
 from io import TextIOWrapper
 
 from rest_framework import status
 from rest_framework.response import Response
 from rest_framework.exceptions import ValidationError
+from django.core.files.base import ContentFile
 
 from mathesar.serializers import DataFileSerializer
-from mathesar.imports.csv import create_table_from_csv, get_sv_dialect
+from mathesar.imports.csv import get_sv_dialect
 from mathesar.errors import InvalidTableError
 
 
-def create_table_from_datafile(data):
-    name = data['name']
-    schema = data['schema']
-    if len(data['data_files']) == 1:
-        data_file = data['data_files'][0]
-    else:
-        raise ValidationError({'data_files': 'Multiple data files are unsupported'})
-
-    table = create_table_from_csv(data_file, name, schema)
-    return table
+def create_datafile(request, data):
+    header = data.get('header', True)
 
+    if 'paste' in data and 'file' in data:
```

Comment:
I should have caught this in previous PRs, but validation like this belongs in the serializer's `validate_<blah>` and `validate` methods. See:
https://www.django-rest-framework.org/api-guide/serializers/#field-level-validation
https://www.django-rest-framework.org/api-guide/serializers/#object-level-validation